sharktank/tests/utils/iree_test.py

```diff
@@ -115,3 +119,139 @@ def roundtrip(iree_devices: list[iree.runtime.HalDevice]):
+        mlir_path = tmp_path / "model.mlir"
+        vmfb_path = tmp_path / "model.vmfb"
+
+        vmfb_bytes = compile_torch_module_to_iree(
+            model,
+            example_args=(example_input,),
+            compile_args=COMPILE_FLAGS,
+            save_mlir_to=mlir_path,
+            save_vmfb_to=vmfb_path,
+        )
+
+        assert len(vmfb_bytes) > 100
+
+        assert mlir_path.exists()
+        assert mlir_path.stat().st_size > 0
+
+        assert vmfb_path.exists()
+        assert vmfb_path.stat().st_size > 0
+
+
+class TestAdaptTorchModuleToIree:
+    """Tests for adapt_torch_module_to_iree."""
+
+    def test_basic_loading_and_execution(self):
+        """Test that loaded module executes and produces correct output shape."""
+        model = SimpleModel()
+        example_input = torch.randn(2, 32)
+
+        iree_module = adapt_torch_module_to_iree(
+            model,
+            example_args=(example_input,),
+            device="local-sync",
+            compile_args=COMPILE_FLAGS,
+        )
+
+        result = iree_module.forward(example_input)
+        assert isinstance(result, torch.Tensor)
+        assert result.shape == (2, 32)
+        assert not torch.isnan(result).any()
+
+    def test_output_matches_torch(self):
+        """Test that IREE output matches torch output."""
+        torch.manual_seed(42)
+        model = SimpleModel()
+        model.eval()
+        example_input = torch.randn(2, 32)
+
+        torch_output = model(example_input)
+        iree_module = adapt_torch_module_to_iree(
+            model,
+            example_args=(example_input,),
+            device="local-sync",
+            compile_args=COMPILE_FLAGS,
+        )
+        iree_output = iree_module.forward(example_input)
+
+        torch.testing.assert_close(iree_output, torch_output, rtol=1e-4, atol=1e-4)
+
+    def test_multi_output_model(self):
+        """Test model with multiple outputs."""
+        model = MultiOutputModel()
+        example_input = torch.randn(2, 32)
+
+        iree_module = adapt_torch_module_to_iree(
+            model,
+            example_args=(example_input,),
+            device="local-sync",
+            compile_args=COMPILE_FLAGS,
+        )
+
+        result = iree_module.forward(example_input)
+        assert isinstance(result, tuple)
+        assert len(result) == 2
+        assert result[0].shape == (2, 32)
+        assert result[1].shape == (2, 32)
+
+
+class TestOneshotCompileAndRun:
+    """Tests for oneshot_iree_run."""
+
+    def test_basic_oneshot(self):
+        """Test basic one-shot execution."""
+        model = SimpleModel()
+        example_input = torch.randn(2, 32)
+
+        result = oneshot_iree_run(
+            model,
+            args=(example_input,),
+            device="local-sync",
+            compile_args=COMPILE_FLAGS,
+        )
+        assert isinstance(result, torch.Tensor)
+        assert result.shape == (2, 32)
+        assert not torch.isnan(result).any()
```

Comment:
This check depends on non-deterministic values.

Reply:
Uses integers and deterministic_random_seed now, done.